java-checks/src/main/java/org/sonar/java/checks/security/IntegerToHexStringCheck.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.checks.security;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import org.sonar.check.Rule;
+import org.sonar.java.checks.methods.AbstractMethodDetection;
+import org.sonar.java.matcher.MethodMatcher;
+import org.sonar.java.matcher.TypeCriteria;
+import org.sonar.plugins.java.api.tree.BinaryExpressionTree;
+import org.sonar.plugins.java.api.tree.ExpressionTree;
+import org.sonar.plugins.java.api.tree.LiteralTree;
+import org.sonar.plugins.java.api.tree.MethodInvocationTree;
+import org.sonar.plugins.java.api.tree.Tree;
+
+@Rule(key = "S4425")
+public class IntegerToHexStringCheck extends AbstractMethodDetection {
+
+  private static final MethodMatcher APPEND_MATCHER = MethodMatcher.create()
+    .typeDefinition(TypeCriteria.subtypeOf("java.lang.AbstractStringBuilder"))
+    .name("append")
+    .addParameter(TypeCriteria.is("java.lang.String"));
+
+  private static final MethodMatcher PRINT_MATCHER = MethodMatcher.create()
+    .typeDefinition(TypeCriteria.subtypeOf("java.io.PrintStream"))
+    .name("print")
+    .addParameter(TypeCriteria.is("java.lang.String"));
+
+  @Override
+  protected List<MethodMatcher> getMethodInvocationMatchers() {
+    return Collections.singletonList(
+      MethodMatcher.create()
+        .typeDefinition(TypeCriteria.is("java.lang.Integer"))
+        .name("toHexString")
+        .addParameter(TypeCriteria.is("int")));
+  }
+
+  @Override
+  protected void onMethodInvocationFound(MethodInvocationTree method) {
+    if (isArgumentAppended(method) && typeIsByte(method.arguments().get(0))) {
+      reportIssue(method.methodSelect(), "Use String.format( \"%02X\", ...) instead.");
+    }
+  }
+
+  private boolean isArgumentAppended(MethodInvocationTree method) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Make "isArgumentAppended" a "static" method. ([squid:S2325](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.java%3Ajava&pullRequest=2010&issues=AWNEzaTfOa7A50ZeJHLt&open=AWNEzaTfOa7A50ZeJHLt)

[ISSUE_KEY]: AWNEzaTfOa7A50ZeJHLt